Mathlib/Analysis/Normed/Unbundled/AlgNormOfAut.lean

```diff
@@ -0,0 +1,144 @@
+/-
+Copyright (c) 2023 María Inés de Frutos-Fernández. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: María Inés de Frutos-Fernández
+-/
+import Mathlib.Analysis.Normed.Unbundled.FiniteExtension
+import Mathlib.LinearAlgebra.FreeModule.Finite.Matrix
+
+/-!
+# algNormOfAlgEquiv and algNormOfAut
+
+Let `K` be a nonarchimedean normed field and `L/K` be a finite algebraic extension. In the comments,
+`‖ ⬝ ‖` denotes any power-multiplicative `K`-algebra norm on `L` extending the norm on `K`.
+
+## Main Definitions
+
+* `IsNonarchimedean.algNormOfAlgEquiv` : given `σ : L ≃ₐ[K] L`, the function `L → ℝ` sending `x : L`
+to `‖ σ x ‖` is a `K`-algebra norm on `L`.
+* `IsNonarchimedean.algNormOfAut` : the function `L → ℝ` sending `x : L` to the maximum of `‖ σ x ‖`
+over all `σ : L ≃ₐ[K] L` is a `K`-algebra norm on `L`.
+
+## Main Results
+* `IsNonarchimedean.isPowMul_algNormOfAlgEquiv` : `algNormOfAlgEquiv` is power-multiplicative.
+* `IsNonarchimedean.isNonarchimedean_algNormOfAlgEquiv` : `algNormOfAlgEquiv` is nonarchimedean.
+* `IsNonarchimedean.algNormOfAlgEquiv_extends` : `algNormOfAlgEquiv` extends the norm on `K`.
+* `IsNonarchimedean.isPowMul_algNormOfAut` : `algNormOfAut` is power-multiplicative.
+* `IsNonarchimedean.isNonarchimedean_algNormOfAut` : `algNormOfAut` is nonarchimedean.
+* `IsNonarchimedean.algNormOfAut_extends` : `algNormOfAut` extends the norm on `K`.
+
+## References
+* [S. Bosch, U. Güntzer, R. Remmert, *Non-Archimedean Analysis*][bosch-guntzer-remmert]
+
+## Tags
+
+algNormOfAlgEquiv, algNormOfAut, norm, nonarchimedean
+-/
+
+open scoped NNReal
+
+noncomputable section
+
+variable {K : Type*} [NormedField K] {L : Type*} [Field L] [Algebra K L]
+  (h_fin : FiniteDimensional K L)
```

Comment:
Why is this not a typeclass assumption?

Reply:
I have made it a typeclass assumption and instead made `L` explicit in the definition of `algNormOfAut`.